lib/App/index.js

```diff
@@ -507,71 +507,71 @@ class App {
   _validateFlowCard(card, errorPath) {
     if (Array.isArray(card.args) === false) return;
 
-    const checkTitleFormatted = (titleFormatted, errorPath) => {
-
-      let foundFirstDevice = false;
-
-      // Filter to valid arguments for flow card titles
-      const args = card.args.filter(arg => {
-        if( foundFirstDevice ) return true;
-        if(arg.type !== 'device') return true;
-        if(!arg.filter) return true;
-        const query = querystring.parse(arg.filter);
-        if(!query.driver_id && !query.driverId) return true;
-        foundFirstDevice = true;
-        return false;
-      });
-
-      if (args.length === 0) return;
-
-      const argsPresent = args.reduce((obj, arg) => {
-        obj[arg.name] = false;
-        return obj;
-      }, {});
-
-      if( card.droptoken )
-        argsPresent.droptoken = false;
-
-      // Match any characters between `[[` and `]]`
-      const argsMatches = titleFormatted.match(/\[\[(.*?)\]\]/gm);
-      if(argsMatches === null)
-          throw Error(`Missing all args in ${errorPath}`);
-
-      argsMatches.forEach(argMatch => {
-        const argName = argMatch.substring(2, argMatch.length-2);
-        if( typeof argsPresent[argName] === 'undefined' )
-          throw Error(`Invalid [[${argName}]] in ${errorPath}.titleFormatted`);
-
-        if( argsPresent[argName] === true )
-          throw Error(`Duplicate [[${argName}]] in ${errorPath}.titleFormatted`);
+    let foundFirstDevice = false;
+
+    // Filter to valid arguments for flow card titles
+    const filteredCardArgs = card.args.filter(arg => {
+      if( foundFirstDevice ) return true;
+      if(arg.type !== 'device') return true;
+      if(!arg.filter) return true;
+      const query = querystring.parse(arg.filter);
+      if(!query.driver_id && !query.driverId) return true;
+      foundFirstDevice = true;
+      return false;
+    });
```

Comment:
I believe we could simply do the following, not sure why `foundFirstDevice` is checked. Maybe its something related to performance?

```suggestion
    // Filter to valid arguments for flow card titles
    const filteredCardArgs = card.args.filter(arg => {
      const query = querystring.parse(arg.filter);

      if (arg.type === 'device' && (query.driver_id || query.driverId)) {
        return false;
      }

      return true;
    });
```

Reply:
Not sure but I remember some weird behaviour with compose adding a second device arg. Don't know if related.

Reply:
I'm pretty sure its because you can have multiple device arguments.

Reply:
`If the card has more than one device fields, the other fields will behave like an autocomplete-like argument, which show devices paired in your app.`